recipes/gtsam/all/conanfile.py

```diff
@@ -163,3 +163,6 @@ def package_info(self):
         self.cpp_info.libs = tools.collect_libs(self)
         if self.settings.os == "Windows" and tools.Version(self.version) >= "4.0.3":
             self.cpp_info.system_libs = ["dbghelp"]
+        # GTSAM only needs a subset of boost, so we manually specify the requirements list.
+        self.cpp_info.requires = ["eigen::eigen"]
+        self.cpp_info.requires.extend(['boost::serialization', 'boost::filesystem', 'boost::system', 'boost::timer','boost::thread', 'boost::date_time'])
```

Comment:
```suggestion
        self.cpp_info.requires = ["eigen::eigen"]
        self.cpp_info.requires.extend(["boost::serialization", "boost::filesystem", "boost::system", "boost::timer", "boost::thread", "boost::date_time"])
        if self.options.with_TBB:
            self.cpp_info.requires.append("tbb::tbb")
        if self.options.install_geographiclib:
            self.cpp_info.requires.append("geographiclib::geographiclib")
```

Moreover, you should:
- add `required_conan_version = ">=1.32.0"`
- check in `validate()` that options of boost recipe related to required components are properly set.

Reply:
Do you mean checking that we're not using boost without serialization of any of the required modules?